scheduler.go

```diff
@@ -70,12 +70,9 @@ func (s *Scheduler) SetLocation(newLocation *time.Location) {
 
 // scheduleNextRun Compute the instant when this Job should run next
 func (s *Scheduler) scheduleNextRun(j *Job) error {
-	now := s.time.Now().In(s.loc)
-	if j.lastRun == s.time.Unix(0, 0) {
-		j.lastRun = now
-	}
+	now := s.time.NowRoundedDownToSeconds(s.loc)
 
-	if j.nextRun.After(now) {
+	if j.nextRun.Second() > now.Second() {
```

Comment:
Any reason for changing this?

Reply:
@arjunmahishi it's because .After() considers nanoseconds while our implementation of .shouldRun() only considers seconds (it mutates the time using .Unix()), so we have two different time precisions within the scheduler.

I believe we should stick to using Unix here as well to compare the whole timestamp... One problem with this current implementation is that if we ommit comparing minutes and hours. Consider the following

nextRun = 10:20:35
now = 10:30:30

NextRun is not after now, but can you spot the seconds issue?

Reply:
That's a good call out. Essentially we just want to compare full time stamps rounded to seconds. I'll udpate.